MantidQt/CustomInterfaces/src/EnggDiffraction/EnggDiffractionViewQtGUI.cpp

```diff
@@ -1261,6 +1340,69 @@ void EnggDiffractionViewQtGUI::setListWidgetBank(int idx) {
   selectBank->setCurrentRow(idx);
 }
 
+void MantidQt::CustomInterfaces::EnggDiffractionViewQtGUI::setPeakPick() {
+  auto bk2bk =
+      FunctionFactory::Instance().createFunction("BackToBackExponential");
+  auto bk2bkFunc = boost::dynamic_pointer_cast<IPeakFunction>(bk2bk);
+  // set the peak to BackToBackExponential function
+  setPeakPicker(bk2bkFunc);
+  setPeakPickerEnabled(true);
+}
+
+void MantidQt::CustomInterfaces::EnggDiffractionViewQtGUI::addPeakToList() {
+
+  if (m_peakPicker->isEnabled()) {
```

Comment:
In next iterations this is the kind of code that should be moved into the presenter. This type of actions can be handled with a notification to the presenter which takes care of saving files, etc.